pkg/convert/convert.go

```diff
@@ -76,91 +77,229 @@ func RegionsCsvToResources(resourcesFile string) ([]map[string]interface{}, erro
+		file, err := os.Open(appInterfaceConfigmapFile)
+		if err != nil {
+			return err
+		}
 
-	// Machine types
-	i := 0
-	_, err = file.Write([]byte("---\napiVersion: v1\nkind: ConfigMap\nmetadata:\n  " +
-		"annotations:\n    " + "qontract.recycle: \"true\"\n  " + "name: cloud-resources-config\ndata:\n  instance-types.yaml: |\n    instance_types:\n"))
-	if err != nil {
-		panic(err)
-	}
-	for _, resource := range resources {
-		if resource[ResourceType] == "compute.node" {
-			// Create block for this machine type
-			id := helper.AssignStringValue(resource["id"])
-			machineTypeString := []byte(fmt.Sprintf(
-				"    - id: %s\n"+
-					"      name: %s\n"+
-					"      cloud_provider_id: %s\n"+
-					"      cpu_cores: %s\n"+
-					"      memory: %s\n"+
-					"      category: %s\n"+
-					"      size: %s\n"+
-					"      generic_name: %s\n",
-				id, resource[Name].(string), resource[CloudProviderId].(string), resource[CpuCores].(string),
-				resource[Memory].(string), resource[Category].(string), resource[SizePretty].(string),
-				resource[GenericName].(string)))
-			_, err = file.Write(machineTypeString)
-			if err != nil {
-				panic(err)
+		fmt.Println(fmt.Sprintf("Rewriting app-interface configmap (%s) with new data supplied from CSV...\n", appInterfaceConfigmapFile))
+
+		scanner := bufio.NewScanner(file)
+		finalText := ""
+		newConfigmap := ""
+		for i, resource := range resources {
+			if resource[ResourceType] == "compute.node" {
+				if i == 0 {
+					continue
+				}
+
+				id := helper.AssignStringValue(resource["id"])
+
+				data := []byte(fmt.Sprintf(
+					"    - id: %s\n"+
+						"      name: %s\n"+
+						"      cloud_provider_id: %s\n"+
+						"      cpu_cores: %s\n"+
+						"      memory: %s\n"+
+						"      category: %s\n"+
+						"      size: %s\n"+
+						"      generic_name: %s\n",
+					id, resource[Name].(string), resource[CloudProviderId].(string), resource[CpuCores].(string),
+					resource[Memory].(string), resource[Category].(string), resource[SizePretty].(string),
+					resource[GenericName].(string)))
+
+				newConfigmap += string(data)
 			}
-			i++
 		}
+
+		skip := false
+		for scanner.Scan() {
+			text := scanner.Text()
+			if strings.Contains(text, "instance_types:") {
+				finalText += fmt.Sprintf("%s\n", text)
+				finalText += fmt.Sprintf("%s", newConfigmap)
+				skip = true
+			} else if !skip {
+				finalText += fmt.Sprintf("%s\n", text)
+			} else if strings.Contains(text, "cloud-regions.yaml: |") {
+				skip = false
+				finalText += fmt.Sprintf("%s\n", text)
+			}
+		}
+
+		defer file.Close()
+
+		if err := os.Truncate(appInterfaceConfigmapFile, 0); err != nil {
+			return err
+		}
+
+		newFile, err := os.OpenFile(appInterfaceConfigmapFile, os.O_APPEND|os.O_WRONLY|os.O_CREATE, 0600)
+		if err != nil {
+			return err
+		}
+
+		_, err = newFile.Write([]byte(finalText))
+		if err != nil {
+			return err
+		}
+
+		defer newFile.Close()
```

Comment:
If an error occurs writing the file, the defer is never called and the `newFile.Close()` will never happen.
I think the `defer` should be moved few lines above:
```suggestion
		newFile, err := os.OpenFile(appInterfaceConfigmapFile, os.O_APPEND|os.O_WRONLY|os.O_CREATE, 0600)
		if err != nil {
			return err
		}
		defer newFile.Close()

		_, err = newFile.Write([]byte(finalText))
		if err != nil {
			return err
		}
```